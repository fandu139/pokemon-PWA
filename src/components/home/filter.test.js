import React from "react";
import Render from 'react-test-renderer';
import Filter, { getFilterData } from './filter';

describe('Home Pokemon', () => {

  const dataCategoryDummy = ["Grass", "Poison", "Fire", "Flying", "Water", "Bug"]
  const dataDummy = [
    {
      "id": "UG9rZW1vbjowMDE=",
      "name": "Bulbasaur",
      "number": "001",
      "classification": "Seed PokÃ©mon",
      "types": [
        "Grass",
        "Poison"
      ],
      "resistant": [
        "Water",
        "Electric",
        "Grass",
        "Fighting",
        "Fairy"
      ],
      "attacks": {
        "fast": [
          {
            "name": "Tackle",
            "type": "Normal",
            "damage": 12
          },
          {
            "name": "Vine Whip",
            "type": "Grass",
            "damage": 7
          }
        ],
        "special": [
          {
            "name": "Power Whip",
            "type": "Grass",
            "damage": 70
          },
          {
            "name": "Seed Bomb",
            "type": "Grass",
            "damage": 40
          },
          {
            "name": "Sludge Bomb",
            "type": "Poison",
            "damage": 55
          }
        ]
      },
      "weaknesses": [
        "Fire",
        "Ice",
        "Flying",
        "Psychic"
      ],
      "fleeRate": 0.1,
      "maxCP": 951,
      "evolutions": [
        {
          "id": "UG9rZW1vbjowMDI=",
          "number": "002"
        },
        {
          "id": "UG9rZW1vbjowMDM=",
          "number": "003"
        }
      ],
      "evolutionRequirements": {
        "amount": 25,
        "name": "Bulbasaur candies"
      },
      "maxHP": 1071,
      "image": "https://img.pokemondb.net/artwork/bulbasaur.jpg"
    }
  ]

  test('Home Filter SnapShoot', () => {
    const props = { 
      dataCategory: dataCategoryDummy,
      data: dataDummy,
      setData: jest.fn(),
      press: jest.fn(),
      dataFilter: dataDummy, 
      setDataFilter: jest.fn()
    };

    const component = Render.create(<Filter {...props} />).toJSON();
    expect(component).toMatchSnapshot();
  });

  test('Function getFilterData', () => {
    const props = { 
      item: dataDummy[0], 
      setData: () => { return true }, 
      dataFilter: dataDummy
    };
    const result = getFilterData(props)
    expect(result).toBe(true)
  });
});
