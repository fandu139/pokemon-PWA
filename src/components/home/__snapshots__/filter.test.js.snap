// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home Pokemon Home Filter SnapShoot 1`] = `
<div
  className="sc-bdVaJa eTWqhe"
>
  <div
    className="App"
  >
    <header
      className="App-button"
    >
      <div
        className="btn-toolbar"
        role="toolbar"
      >
        <button
          className="btn btn-success btn-sm"
          disabled={false}
          onClick={[Function]}
          type="button"
        >
          Grass
        </button>
      </div>
    </header>
  </div>
  <div
    className="App"
  >
    <header
      className="App-button"
    >
      <div
        className="btn-toolbar"
        role="toolbar"
      >
        <button
          className="btn btn-success btn-sm"
          disabled={false}
          onClick={[Function]}
          type="button"
        >
          Poison
        </button>
      </div>
    </header>
  </div>
  <div
    className="App"
  >
    <header
      className="App-button"
    >
      <div
        className="btn-toolbar"
        role="toolbar"
      >
        <button
          className="btn btn-success btn-sm"
          disabled={false}
          onClick={[Function]}
          type="button"
        >
          Fire
        </button>
      </div>
    </header>
  </div>
  <div
    className="App"
  >
    <header
      className="App-button"
    >
      <div
        className="btn-toolbar"
        role="toolbar"
      >
        <button
          className="btn btn-success btn-sm"
          disabled={false}
          onClick={[Function]}
          type="button"
        >
          Flying
        </button>
      </div>
    </header>
  </div>
  <div
    className="App"
  >
    <header
      className="App-button"
    >
      <div
        className="btn-toolbar"
        role="toolbar"
      >
        <button
          className="btn btn-success btn-sm"
          disabled={false}
          onClick={[Function]}
          type="button"
        >
          Water
        </button>
      </div>
    </header>
  </div>
  <div
    className="App"
  >
    <header
      className="App-button"
    >
      <div
        className="btn-toolbar"
        role="toolbar"
      >
        <button
          className="btn btn-success btn-sm"
          disabled={false}
          onClick={[Function]}
          type="button"
        >
          Bug
        </button>
      </div>
    </header>
  </div>
</div>
`;
